src/giftCards/GiftCardSettings/utils.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  GiftCardSettingsExpiryTypeEnum,
+  GiftCardSettingsUpdateInput
+} from "@saleor/types/globalTypes";
+
+import { GiftCardSettingsFormData } from "./types";
+
+export const getGiftCardSettingsInputData = ({
+  expiryPeriodActive,
+  expiryPeriodType,
+  expiryPeriodAmount
+}: Pick<
+  GiftCardSettingsFormData,
+  "expiryPeriodActive" | "expiryPeriodType" | "expiryPeriodAmount"
+>): GiftCardSettingsUpdateInput => ({
```

Comment:
I think this could look a bit cleaner
```suggestion
>): GiftCardSettingsUpdateInput => {
  if (expiryPeriodActive) {
    return {
      expiryType: GiftCardSettingsExpiryTypeEnum.EXPIRY_PERIOD,
      expiryPeriod: {
        type: expiryPeriodType,
        amount: expiryPeriodAmount
      }
    };
  }

  return {
    expiryType: GiftCardSettingsExpiryTypeEnum.NEVER_EXPIRE
  };
};
```

Especially since we always keep a default value for the select in the form initial data, right? So the case when we chose expiry period, and have no amount (like the check in line 20 here) will never happen, therefore I don't think we need that check

Reply:
No, there is a possibility when user may wipe out amount input and then using backend side validation it returns error.